ui/ops/src/util/array.js

```diff
@@ -0,0 +1,66 @@
+import {nextTick} from 'vue';
+
+/**
+ * Iterates over large arrays asynchronously
+ *
+ * @param {Array} array
+ * @param {function} cb
+ * @param {boolean} batched
+ */
+export async function iterateLargeArray(array, cb, batched = false) {
+  if (batched) {
+    for (const batch of array) {
+      batch.forEach(cb);
+      await nextTick();
+    }
+    return;
+  }
+
+  for (const ele of array) {
+    cb(ele);
+    await nextTick();
+  }
+}
+
+/**
+ * Maps large arrays asynchronously
+ *
+ * @param {Array} array
+ * @param {function} cb
+ * @param {boolean} batched
+ * @return {Array}
+ */
+export async function mapLargeArray(array, cb, batched = false) {
```

Comment:
As with the iterate func, I'm not sure this needs to exist.

```js
const res = arr.map(fn);
// becomes
const res = batch(arr, 10).map((b) => b.map(fn)).flat();
// or async
const res = [];
for (const b of batch(arr, 10)) {
  res.push(...b.map(fn));
  await nextTick();
}
```

Clear what it does, I think.